src/LanguageServer/Impl/Diagnostics/DiagnosticsService.cs

```diff
@@ -36,11 +36,18 @@ private sealed class DocumentDiagnostics {
             public bool Changed { get; set; }
 
             public void Clear(DiagnosticSource source) {
-                Changed = Entries.Count > 0;
+                if (!_entries.TryGetValue(source, out _)) {
```

Comment:
if there is no entry in _entries, I believe nothing could considered changed. but if we want to make Changed always true, then I can revert it back to original code.